src/drt/src/dr/FlexDR_maze.cpp

```diff
@@ -2338,8 +2338,11 @@ void FlexDRWorker::routeNet_postAstarWritePath(
         FlexMazeIdx mi(startX, startY, currZ);
         auto via = getTech()->getLayer(startLayerNum + 1)->getDefaultViaDef();
         if (net->getFrNet()->getNondefaultRule()
-            && net->getFrNet()->getNondefaultRule()->getPrefVia(currZ))
-          via = net->getFrNet()->getNondefaultRule()->getPrefVia(currZ);
+            && net->getFrNet()->getNondefaultRule()->getPrefVia(
```

Comment:
Extracting `net->getFrNet()->getNondefaultRule()` into its own variable will help the readability of this section.